src/custom-element.js

```diff
@@ -566,6 +566,11 @@ function createBaseCustomElementClass(win) {
 
       /** @private {?./layout-delay-meter.LayoutDelayMeter} */
       this.layoutDelayMeter_ = null;
+
+      if (this['whenUpgradedToCustomElement']) {
+        this['whenUpgradedToCustomElement'](this);
+        this['whenUpgradedToCustomElement'] = null;
```

Comment:
set element['promise'] to null. To confirm when I do `return element['promise']` it is not returning a reference to the promise? So setting `element['promise'] = null` doesn't affect the returned promise from `#whenUpgradedToCustomElement` function.